src/Object/BaseObject.php

```diff
@@ -54,13 +53,21 @@ public function __isset($name)
     }
 
     /**
+     * @param $name
+     */
+    public function __unset($name)
+    {
+        unset($this->data[$name]);
+    }
+
+    /**
      * @param array
      * @return $this
      */
     public function setData(array $data)
     {
         foreach ($data as $key => $value) {
-            $this->data[$key] = $value;
+            $this->{$key} = $value;
```

Comment:
You're using $this->property here, but in unset and __get you're using $this->data?